blogs/2020/02/14/optimizing-ci.md

```diff
@@ -0,0 +1,117 @@
+---
+Order: 53
+TOCTitle: Improving CI Build Times
+PageTitle: How Visual Studio Code leaverages Azure Pipelines Artifact Caching Tasks to improve CI
+MetaDescription: How Visual Studio Code leaverages Azure Pipelines Artifact Caching Tasks to improve CI
+MetaSocialImage: /assets/blogs/2020/02/14/TODO
+Date: 2020-02-14
+ShortDescription: How Visual Studio Code leaverages Azure Pipelines Artifact Caching Tasks to improve CI
+Author: Ethan Dennis, João Moreno
+---
+# Improving CI Build Times
+
+February 14, 2020 by Ethan Dennis, [@erdennis13](https://twitter.com/erdennis13) and João Moreno, [@joaomoreno](https://twitter.com/joaomoreno)
+
+Visual Studio Code is a large project with lots of moving parts and an active participant list. We have [shown](https://code.visualstudio.com/blogs/2018/09/12/engineering-with-azure-pipelines) how we actively use Azure Pipelines to keep up with good engineering practices by maintaining our build and continuous integration infrastructure. In this blog post, we’ll talk about how we used the [Azure Pipelines Artifact Caching Tasks](https://github.com/Microsoft/azure-pipelines-artifact-caching-tasks) to dramatically reduce our CI build times.
+
+We described in an [earlier blog post](https://medium.com/crawl-walk-sprint/reducing-vs-code-ci-build-times-by-33-dbb1715b5028) how we reduced CI build times by 33%. This was accomplished by using custom build tasks that cache the node modules that VS Code consumes instead of resolving the packages at build time. While we were happy with this performance boost, we wanted to see how much further we could push the caching tasks that we built.
+
+When we last talked about our CI engineering our target platforms spanned Windows, macOS, and Linux. As of today, VS Code targets a much more diverse set of platforms, such as ARM64 and Alpine Linux for its remote server components. In total, we have eight different targets that all share common build steps. This post outlines how we leveraged the caching tasks to reduce CI duplication and further improve our build times.
+
+## Room for Improvement
+
+So, what exactly were the common steps across all build jobs? Each build target has a job that follows a similar set of steps. At a very high level, each job must:
+
+1. Restore Node packages
+2. Lint the code
+3. Compile the code
```

Comment:
Suggest make it more clear that in VS Code we compile from TypeScript to JavaScript and therefore would describe this step as: Compile the code from TypeScript to JavaScript.